hoomd/hpmc/ComputeSDF.h

```diff
@@ -264,27 +269,64 @@ template<class Shape> void ComputeSDF<Shape>::zeroHistogram()
     // Zero the histogram
     for (size_t i = 0; i < m_hist.size(); i++)
         {
-        m_hist[i] = 0;
+        m_hist[i] = 0.0;
         }
     }
 
+template<class Shape> Scalar ComputeSDF<Shape>::getMaxInteractionDiameter()
+    {
+    Scalar max_core_diameter = m_mc->getMaxCoreDiameter();
+    Scalar max_r_cut_patch(0.0);
+
+    if (m_mc->getPatchEnergy())
+        {
+        const std::vector<param_type, hoomd::detail::managed_allocator<param_type>>& params
+            = m_mc->getParams();
+        for (unsigned int typ_i = 0; typ_i < m_pdata->getNTypes(); typ_i++)
+            {
+            Shape shape_i(quat<Scalar>(), params[typ_i]);
+            Scalar r_cut_patch_i = m_mc->getPatchEnergy()->getRCut()
+                                   + 0.5 * m_mc->getPatchEnergy()->getAdditiveCutoff(typ_i);
```

Comment:
```suggestion
                                   + m_mc->getPatchEnergy()->getAdditiveCutoff(typ_i);
```
This method computes a diameter, not a radius. It needs to find the largest distance between two particles that would be interacting. The largest distance is the correct value to use when computing the system wide extra ghost width (see the call to `setExtraGhostWidth`). It is an over-conservative estimate when computing `extra_width` as we could use the known additive cutoff for type `i` to reduce the distance somewhat. *However*, I don't think that is necessary as it makes the code even more complex, and it is not clear how much performance benefit there would be. Let's make a note of it and can revisit if anyone ever comes to me with an additive cutoff system with the SDF performance as the main bottleneck.